python/ray/data/preprocessor.py

```diff
@@ -249,16 +249,28 @@ def _transform(self, dataset: Dataset) -> Dataset:
 
         # Our user facing batch format should only be pandas or numpy, other
         # formats {arrow, simple} are internal.
+        kwargs = self._get_transform_config()
         if transform_type == "pandas":
-            return dataset.map_batches(self._transform_pandas, batch_format="pandas")
+            return dataset.map_batches(
+                self._transform_pandas, batch_format="pandas", **kwargs
```

Comment:
can we narrow this function down to each field, rather than an arbitrary kwargs dict ?
there're only `batch_size ` , `compute` and `batch_format` in `map_batches` that likely require explicit value, where `**` is used for `ray_remote_args` that ideally we don't want to be mixed with param values above.

```python
def map_batches(
  |   | self,
  |   | fn: BatchUDF,
  |   | *,
  |   | batch_size: Optional[int] = 4096,
  |   | compute: Optional[Union[str, ComputeStrategy]] = None,
  |   | batch_format: Literal["default", "pandas", "pyarrow", "numpy"] = "default",
  |   | fn_args: Optional[Iterable[Any]] = None,
  |   | fn_kwargs: Optional[Dict[str, Any]] = None,
  |   | fn_constructor_args: Optional[Iterable[Any]] = None,
  |   | fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
  |   | **ray_remote_args,
  |   | ) -> "Dataset[Any]":
```

Reply:
The point of the `**kwargs` passthrough is to make this friendly to current and future `.map_batches()` arguments, where subclasses of `Preprocessor` can opt in to passing any of these args, including custom preprocessors. Otherwise, we'll need update the base `Preprocessor` whenever exposing a new parameter, and devs/advanced users will either be blocked from using that parameter until the next release or will have to override `Preprocessor._transform()`. Explicitly enumerating each potential field that may or may not exist on the preprocessor will also be more complex than this kwarg passthrough.

>where ** is used for ray_remote_args that ideally we don't want to be mixed with param values above.

What if we (or a user) has a preprocessor that we (they) want to run on a GPU, or it's mostly I/O-bound so we (they) to request a fractional CPU, or we (they) want to specify custom retry logic? We _do_ want to expose these.

Since this kwarg passthrough isn't exposed to any users except for advanced users implementing custom preprocessors, is this really an issue? I feel like the future-proofing + simplicity advantages outweigh the disadvantages.

Reply:
I'll take a stab at blowing these out into `batch_size`, `compute`, `batch_format`, and `ray_remote_args` getters that can be overridden, but I don't know if it will be a net benefit.

Reply:
Another disadvantage of making each of these kwargs explicit is that supporting `batch_size=None` is more difficult, since we need to delineate between a preprocessor that doesn't set `batch_size` vs a preprocessor that wants to disable batching with `batch_size=None`. So we either need some other "not specified" indicator or start duplicating default values. which isn't great.

Reply:
ah i mean it's less error prone for us to explicitly call with args  
`dataset.map_batches(self._transform_numpy, batch_format="numpy", batch_size={field}, **kwargs)` since technically we can also hide the `batch_format` kwarg completely

Reply:
Hmm what error do you see arising there, could you give an example?

>since technically we can also hide the batch_format kwarg completely

`batch_format` is hardcoded since we have generic logic in `Preprocessor` to determine the appropriate batch format, while the rest (including `batch_size`) are determined entirely by the subclasses, so I think that difference in treatment is reasonable.

I can pop `batch_size` out of the kwargs and pass it explicitly, but if we're still passing the rest as a `**kwargs` passthrough, I'm not sure what that would give us.

Reply:
i think i might missed some context you had, or future plans regarding how we handle batch sizes as always being explicit with its value upon calling each `map_batches` is more appealing to me. But let's consolidate the discussion in the API discussion below :)